Dockerfile.ci

```diff
@@ -1184,6 +1166,23 @@ COPY airflow_pre_installed_providers.txt ${AIRFLOW_SOURCES}/
 COPY hatch_build.py ${AIRFLOW_SOURCES}/
 COPY --from=scripts install_airflow.sh /scripts/docker/
 
+# Those are additional constraints that are needed for some extras but we do not want to
```

Comment:
And "UPGRADE_TO_NEWER_DEPENDENCIES" is now set AFTER the cache is prepared - impacting whether final step of installation is done with costraints (UPGRADE_TO_NEWER_DEPENDENCIES = false) or without (UPGRADE_TO_NEWER_DEPENDENCIES set to a random value to trigger cache invalidation).